quic/s2n-quic-core/src/datagram/mod.rs

```diff
@@ -1,7 +1,7 @@
 // Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
 // SPDX-License-Identifier: Apache-2.0
 
-#[cfg(feature = "alloc")]
+#[cfg(all(feature = "alloc", feature = "std"))]
```

Comment:
technically we only need to gate the `impl std::error::Error ...`, rather than the whole `default` module

Reply:
oh I wasn't sure this line would work then: https://github.com/aws/s2n-quic/blob/e283f2e0bb0016b8a8f0968d44ebec80189db16e/quic/s2n-quic-core/src/datagram/default.rs#L53

Reply:
yeah it definitely should. `std::error::Error` is mostly just a marker trait to allow interoperability with other errors (usually by returning `Box<dyn std::error::Error>`)